### ISSUE TITLE
Question about learning rate, betas, and weight decay in Adam optimizer

### ISSUE THREAD

**Issue by Kaminyou** [binli123/dsmil-wsi#37]
I am wondering why `learning rate` is set to **`0.0002`** while your paper did mention that *We use Adam [25] optimizer
with a constant learning rate of 0.0001 to update the model weights during the training*.
Besides, the settings of `betas` and `weight decay` in `Adam` are uncommon.
```python
# arg
parser.add_argument('--weight_decay', default=5e-3, type=float, help='Weight decay [5e-3]')
# adam
optimizer = torch.optim.Adam(milnet.parameters(), lr=args.lr, betas=(0.5, 0.9), weight_decay=args.weight_decay)
```
Is there any reason to set the hyperparameters like these (especially the strange settings of `betas`)?

**Comment by binli123** [comment]
Hi, you could verify that using lr=0,0001, default betas and 0 weight decay has very minor impacts on the results. Also, these settings are not strange at all and are quite common practices for small batch sizes and small datasets.

**Comment by Kaminyou** [comment]
@binli123 Got it. I know the learning rate (and other settings) would have a minor impact on the results. However, I am wondering if you claim you use `0.0001` in the paper (but actually use `0.0002` in implementation) would somehow lead to trustworthiness or not.

**Comment by binli123** [comment]
> @binli123 Got it. I know the learning rate (and other settings) would have a minor impact on the results. However, I am wondering if you claim you use `0.0001` in the paper (but actually use `0.0002` in implementation) would somehow lead to untrustworthiness or not.

I think you need to set lr=0.0001 if you want to reproduce the results. The default value of 0.0002 leads to faster convergence and that is for the convenience of the user. If you have any problems or doubts you could easily set this variable to 0.0001 to verify the results instead of posting in an offensive tone here.

**Comment by Kaminyou** [comment]
@binli123 I see. Thank you very much.

**Comment by binli123** [comment]
> 

No problem at all. I admit that there are some differences between the paper and the actual github repo, but given my bandwidth, I have tried to make this repo as tidy as I can.
